projects/packages/jetpack-mu-wpcom/src/features/post-categories/quick-actions.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Quick actions for the categories list page.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Adds a quick action to change the default post category.
+ *
+ * @param string[] $actions An array of action links to be displayed.
+ * @param WP_Term  $category Category object.
+ *
+ * @return string[] Filtered actions.
+ */
+function wpcom_add_set_default_category_quick_action( $actions, $category ) {
+	if ( ! current_user_can( 'manage_options' ) ) {
+		return $actions;
+	}
+
+	$default_category = (int) get_option( 'default_category' );
+	if ( $category->term_id === $default_category ) {
+		return $actions;
+	}
+
+	$link = add_query_arg(
+		array(
+			'category' => $category->term_id,
+			'action'   => 'wpcom-set-default-category',
+		)
+	);
+	$link = wp_nonce_url( $link, 'wpcom-set-default-category' );
+
+	$actions['set-default'] = sprintf(
+		'<a href="%1$s" aria-label="%2$s">%3$s</a>',
+		esc_url( $link ),
+		/* translators: category name */
+		esc_attr( sprintf( __( 'Set &#8220;%s&#8221; as the default category', 'jetpack-mu-wpcom' ), $category->name ) ),
+		esc_html( __( 'Set as default', 'jetpack-mu-wpcom' ) )
+	);
+	return $actions;
+}
+add_filter( 'category_row_actions', 'wpcom_add_set_default_category_quick_action', 10, 2 );
+
+/**
+ * Changes the default post category.
+ */
+function wpcom_set_default_category() {
+	if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['_wpnonce'] ) ), 'wpcom-set-default-category' ) ) {
+		return;
+	}
+
+	if ( ! current_user_can( 'manage_options' ) ) {
+		return;
+	}
```

Comment:
Should we check this before the nonce for efficiency?

Reply:
I don't see any difference in performance when I measure the execution time with `microtime`.